synapse/models/inet.py

```diff
@@ -330,6 +330,29 @@ def repr(self, valu):
             except UnicodeError:
                 return valu
 
+class HttpCookie(s_types.Str):
+
+    def _normPyStr(self, text):
+
+        text = text.strip()
+        parts = text.split('=', 1)
+
+        name = parts[0].split(';', 1)[0].strip()
+        if len(parts) == 1:
+            return text, {'subs': {'name': name}}
+
+        valu = parts[1].split(';', 1)[0].strip()
+        return text, {'subs': {'name': name, 'value': valu}}
+
+    def getTypeVals(self, valu):
+
+        if isinstance(valu, str):
+            cookies = valu.split(';')
+            cookies = [c.strip() for c in cookies]
+            return [c for c in cookies if c]
+
+        return valu
```

Comment:
Currently non-str types that are not iterable will blow up here, probably should make this a generator?